### PR TITLE
Implement the cluster management UI for Kvrocks controller

### PR DESCRIPTION
**Pull request by CulturalProfessor**
#### Description

This PR introduces a new UI for managing clusters, including the ability to create and delete clusters. The following features and improvements are included:

- [x] **Namespace Sidebar Component**: Reusable `NamespaceSidebar` component listing all namespaces.
- [x] **Cluster Sidebar Component**: Reusable `ClusterSidebar` component listing clusters within a selected.
- [x] **Namespace Page**: Displays clusters within the selected namespace,shows a 404 page if the namespace does not exist.
- [x] **Item Component**: Enhanced to handle both namespaces and clusters, added functionality to delete namespaces and clusters.
- [x] **Card Component**: Reusable card components for creating and displaying cluster details. We can use them for shard and node dashboards also.

#### Screencast

Here is a video demonstration of the new UI features for creating and deleting clusters:

[Screencast from 23-07-24 05:22:09 PM IST.webm](https://github.com/user-attachments/assets/bc4db7aa-8fea-4a7c-a363-4fef88bba77d)

### CODE REVIEW

**Comment by CulturalProfessor** [comment]
@jihuayu @git-hulk please review this pr and suggest any edits.

**Comment by git-hulk** [comment]
@CulturalProfessor Cool, thanks for your great work!

File: webui/public/addIcon.svg
Comment (PragmaTwice) [inline webui/public/addIcon.svg:1]: I'm wondering if we can use some font like fontawesome for all icons?

Comment (CulturalProfessor) [reply webui/public/addIcon.svg:1]: Sure I'll edit the icon set, I was using icons8 as most of the icons are free there :sweat_smile: , do these seem good ?
https://fontawesome.com/icons/circle-plus?f=classic&s=solid
https://fontawesome.com/icons/trash?f=classic&s=solid

**Comment by jihuayu** [comment]
Cool! Thank you! I will have a look tomorrow

**Comment by jihuayu** [comment]
@PragmaTwice Should we add license header for SVG?
```
ERROR the following files don't have a valid license header: 
webui/public/addIcon.svg
webui/public/deleteIcon.svg 
ERROR one or more files does not have a valid license header 
```

**Comment by git-hulk** [comment]
> @PragmaTwice Should we add license header for SVG?
> 
> ```
> ERROR the following files don't have a valid license header: 
> webui/public/addIcon.svg
> webui/public/deleteIcon.svg 
> ERROR one or more files does not have a valid license header 
> ```

We should try our best to add an LICENSE header for all assets unless it's impossible to add that. For those files can be excluded at [licenserc.yaml#L27](https://github.com/apache/kvrocks-controller/blob/unstable/.github/licenserc.yaml#L27). So we should add and include the original license header if they're not made by ourself.

**Comment by CulturalProfessor** [comment]
image
Hey guys I have added the license and used fontawesome icons here, please review them.

File: webui/src/app/lib/actions.ts [hunk 1]
```diff
@@ -1,37 +1,66 @@
...
  * Unless required by applicable law or agreed to in writing,
  * software distributed under the License is distributed on an
  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  * KIND, either express or implied.  See the License for the
  * specific language governing permissions and limitations
- * under the License. 
+ * under the License.
  */
-'use server';
+"use server";
```
Comment (jihuayu) [inline webui/src/app/lib/actions.ts]: Can you move the server part to the client? We need to compile it to a static frontend file.
In our scenario, Client-side rendering is OK; we don't need the runtime features of Next.js.

**Review by jihuayu — COMMENTED** [review]
Can you install the Font Awesome npm package and use `<FontAwesomeIcon icon="fa-solid fa-circle-plus" />` to utilize icons? This will make it easier for using icons later on.

File: webui/public/addIcon.svg
Comment (PragmaTwice) [reply webui/public/addIcon.svg:1]: sure.

File: webui/src/app/lib/actions.ts (hunk 1 above)
Comment (CulturalProfessor) [reply webui/src/app/lib/actions.ts]: should I move api.ts (and some other server components) logic also to client side as it depends on config.yaml?

Comment (jihuayu) [reply webui/src/app/lib/actions.ts]: Yes, We should not depend on backend config. If necessary, we can add frontend config.

We can use relative paths to access the backend. This way, we won't need to depend on `config.yml`. 
During development, we can use Next.js to proxy our requests to the local backend at `127.0.0.1:9379`.
https://dev.to/george3447/setting-up-proxy-in-nextjs-for-dev-enviornment-1nh3


If you find this difficult, we can implement the UI first. And do it at last.

Comment (CulturalProfessor) [reply webui/src/app/lib/actions.ts]: @jihuayu I updated icons to use fontawesome packages, added a proxy for dev environment and changed server components to client side to better suit our usecase.

**Review by jihuayu — COMMENTED** [review]
Good job! Thanks for your work!

**Review by jihuayu — APPROVED** [review]

**Comment by jihuayu** [comment]
Can you update your branch and merge unstable into your branch? @CulturalProfessor

**Comment by CulturalProfessor** [comment]
> Can you update your branch and merge unstable into your branch? @CulturalProfessor

@jihuayu I merged unstable

**Pull request merged**